src/SonarQube.Client/Services/SonarQubeService.cs

```diff
@@ -59,56 +57,60 @@ public bool HasOrganizationsFeature
             }
         }
 
-        public bool IsConnected => this.connection != null;
+        public bool IsConnected => this.sonarqubeClient != null;
 
         public async Task ConnectAsync(ConnectionInformation connection, CancellationToken token)
         {
-            if (this.connection != null)
+            if (this.sonarqubeClient != null)
             {
                 throw new InvalidOperationException("This operation expects the service not to be connected.");
             }
 
-            var ConnectionRequest = new ConnectionRequest
+            var connectionRequest = new ConnectionRequest
             {
                 Authentication = connection.Authentication,
                 ServerUri = connection.ServerUri,
                 Login = connection.UserName,
                 Password = connection.Password
             };
 
-            var credentialsValidationResult = await this.sonarqubeClient.ValidateCredentialsAsync(ConnectionRequest, token);
+            var client = this.sonarqubeClientFactory.Create(connectionRequest);
+
+            var credentialsValidationResult = await client.ValidateCredentialsAsync(token);
             credentialsValidationResult.EnsureSuccess();
 
             if (!credentialsValidationResult.Value.IsValid)
             {
                 throw new Exception("Invalid credentials."); // TODO: Provide better exception
             }
 
-            var versionResult = await this.sonarqubeClient.GetVersionAsync(ConnectionRequest, token);
+            this.sonarqubeClient = client;
```

Comment:
`IsConnected => this.sonarqubeClient != null;`

If `GetVersionAsync` fails, the service will say it's connected, but the version would be null.

Reply:
Yep! Fixed it!